runelite-client/src/main/java/net/runelite/client/plugins/kittentracker/KittenPlugin.java

```diff
@@ -0,0 +1,686 @@
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.Text;
+
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+
+@PluginDescriptor(
+		name = "Kitten"
+)
+
+public class KittenPlugin extends Plugin
+{
+
+	private boolean ready;
+	private Instant kittenSpawned;
+	private Instant catSpawned;
+	private Instant kittenFed;
+	private Instant kittenAttention;
+	private int timeSpendGrowing;
+	private int timeNeglected;
+	private int timeHungry;
+	private static final String CHAT_CAT_STROKE = "That cat sure loves to be stroked.";
+	private static final String CHAT_CAT_BALLOFWOOL = "That kitten loves to play with that ball of wool. I think itis its favourite.";
+	private static final String CHAT_CAT_GROWN = "Your kitten has grown into a healthy cat that can hunt for itself.";
+	private static final String CHAT_CAT_OVERGROWN = "Your cat has grown into a mighty feline, but it will no longer be ableto chase vermin.";
+	private int followerID = 0;
+	private int previousFeline = 0;
+
+
+
+	private boolean cat = false; // npcId 1619-1625
+	private boolean lazycat = false; // npcId 1626-1632
+	private boolean wilycat = false; // npcId 5584-5590
+	private boolean kitten = false; // npcId 5591-5597
+	private boolean overgrown = false; // npcId 5598-5604
+	private boolean nonFeline = false;
+
+
+	@Provides
+	KittenConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(KittenConfig.class);
+	}
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private KittenConfig config;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+
+
+	@Override
+	public void startUp()
+	{
+		clientThread.invokeLater(this::checkForFollower);
+		previousFeline = config.felineId();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		byeFollower();
+	}
+
+	private void checkForFollower()
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		if (playerHasFollower())
+		{
+			newFollower();
+		}
+	}
+
+	@Subscribe
+	public void onVarbitChange(VarbitChanged event)
+	{
+		if (playerHasFollower() & followerID == 0) // player gained a follower
+		{
+			getFollowerID();
+		}
+
+		if (playerHasFollower() & followerID != 0) // follower changed ID
+		{
+			getFollowerID();
+		}
+
+		if (!playerHasFollower() & followerID != 0) // player lost it's follower
+		{
+			byeFollower();
+		}
+
+	}
+
+	private void getFollowerID()
+	{
+		String s = Integer.toBinaryString(client.getVar(VarPlayer.FOLLOWER));
+		while (s.length() < 32)
+		{
+			s = "0" + s;
+		}
+		int previousFollowerID = 0;
```

Comment:
Blank line between closing and opening of logical blocks (various in file)
See `addKittenGrowthBox` for the correct formatting